src/new/wallet.js

```diff
@@ -0,0 +1,441 @@
+    // XXX Update it so we don't have fixed pin/password
+    this.pinCode = '123';
+    this.password = '123';
+  }
+
+  /**
+   * Called when loading the history of transactions.
+   * It is called every HTTP Request to get the history of a set of addresses.
+   * Usually, this is called multiple times.
+   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
+   *
+   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
+   **/
+  onAddressesLoaded(result) {
+    this.emit('more-addresses-loaded', result);
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    if (value) {
+      this.setState(Wallet.SYNCING);
+      wallet.loadAddressHistory(0, GAP_LIMIT).then(() => {
+        this.setState(Wallet.READY);
+      }).catch((error) => {
+        throw error;
+      })
+    } else {
+      // CONNECTING or CLOSED?
+      this.serverInfo = null;
+      this.setState(Wallet.CONNECTING);
+    }
+  }
+
+  getAddressToUse() {
+    return wallet.getAddressToUse();
+  }
+
+  getCurrentAddress() {
+    return wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      this.onNewTx(wsData);
+    }
+  }
+
+  reloadData() {
+    // TODO Reload data?
+    console.log('reloadData');
+  }
+
+  getBalance(tokenUid) {
+    if (!tokenUid && !this.token) {
+      throw Error('Token must be set in the class instance or passed as parameter in the method.')
+    }
+    const uid = tokenUid || this.token.uid;
+    const historyTransactions = this.getTxHistory();
+    return wallet.calculateBalance(Object.values(historyTransactions), uid);
+  }
+
+  getTxHistory() {
+    const data = wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in data ? data['historyTransactions'] : {};
+    return historyTransactions;
+  }
+
+  setState(state) {
+    this.state = state;
+    this.emit('state', state);
+  }
+
+  onNewTx(wsData) {
+    const newTx = wsData.history;
+
+    // TODO we also have to update some wallet lib data? Lib should do it by itself
+    const walletData = wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+
+    let isNewTx = true;
+    if (newTx.tx_id in historyTransactions) {
+      isNewTx = false;
+    }
+
+    wallet.updateHistoryData(historyTransactions, allTokens, [newTx], null, walletData);
+
+    if (isNewTx) {
+      this.emit('new-tx', newTx);
+    } else {
+      this.emit('update-tx', newTx);
+    }
+    return;
+
+    // TODO XXX Uncomment the following block to keep track of the balance.
```

Comment:
Do we need to keep this commented code?

Reply:
I think we will remove it soon. It updates the balance of each token.

Reply:
I agree. I think we should keep it there so @msbrogli can remember about it when doing the bigger refactor.